src/main/kotlin/com/watermelon/context/actions/LoginAction.kt

```diff
@@ -0,0 +1,96 @@
+package com.watermelon.context.actions
+
+import com.intellij.credentialStore.CredentialAttributes
+import com.intellij.ide.BrowserUtil
+import com.intellij.ide.passwordSafe.PasswordSafe
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.ui.Messages
+import kotlinx.serialization.json.*
+import java.net.HttpURLConnection
+import java.net.URL
+
+
+class LoginAction : AnAction() {
+    private val backendUrl = "https://app.watermelontools.com"
+    private fun sendTokenToAPI(token: String): String {
+        val apiUrl = "$backendUrl/api/extension/intellijLogin"
+        val url = URL(apiUrl)
+        val connection = url.openConnection() as HttpURLConnection
+
+        try {
+            connection.requestMethod = "POST"
+            connection.doOutput = true
+            connection.setRequestProperty("Content-Type", "application/json")
+            connection.setRequestProperty("Charset", "UTF-8")
+
+            val payload = """
+            {
+                "token": "$token"
+            }
+        """.trimIndent()
+
+            connection.outputStream.write(payload.toByteArray())
+
+            val responseCode = connection.responseCode
+            return if (responseCode == HttpURLConnection.HTTP_OK) {
+                val jsonResponse = Json.parseToJsonElement(connection.inputStream.reader().readText()).jsonObject
+                val data = jsonResponse["data"]?.jsonObject
+                val id = data?.get("id")?.toString()
+                val email = data?.get("email")?.toString()
+                // Store the tokens
+                val passwordSafe = PasswordSafe.instance
+                passwordSafe.setPassword(CredentialAttributes("WatermelonContext.id"), id)
+                passwordSafe.setPassword(CredentialAttributes("WatermelonContext.email"), email)
+                connection.inputStream.reader().readText()
+            } else {
+                // Handle non-200 HTTP responses
+                "Error: $responseCode"
+            }
+        } finally {
+            connection.disconnect()
+        }
+    }
+
+    override fun actionPerformed(e: AnActionEvent) {
+        // Open webpage
+        BrowserUtil.browse("$backendUrl/intellij")
```

Comment:
To avoid someone else logging in with my account (anyone can just use esteban...@gmail.com) this should evolve towards pasting a key from our dashboard. 

But this is good for now.